src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/DotnetDataCollectionLauncher.cs

```diff
@@ -0,0 +1,193 @@
+            }
+
+            if (EqtTrace.IsVerboseEnabled)
+            {
+                EqtTrace.Verbose("DotnetDataCollectionLauncher: Full path of dotnet.exe is {0}", currentProcessFileName);
+            }
+
+            var dataCollectorAssemblyPath = Path.Combine(currentWorkingDirectory, DataCollectorProcessName);
+            commandLineArguments.Insert(0, dataCollectorAssemblyPath);
+
+            dataCollectorFileName = Path.GetFileNameWithoutExtension(dataCollectorAssemblyPath);
+
+            // .NET core host manager is not a shared host. It will expect a single test source to be provided.
+            var args = "exec";
+
+            // Probe for runtimeconfig and deps file for the test source
+            var runtimeConfigPath = Path.Combine(currentWorkingDirectory, string.Concat(dataCollectorFileName, ".runtimeconfig.json"));
+
+            if (this.fileHelper.Exists(runtimeConfigPath))
+            {
+                var argsToAdd = " --runtimeconfig \"" + runtimeConfigPath + "\"";
+                args += argsToAdd;
+                if (EqtTrace.IsVerboseEnabled)
+                {
+                    EqtTrace.Verbose("DotnetDataCollectionLauncher: Adding {0} in args", argsToAdd);
+                }
+            }
+            else
+            {
+                if (EqtTrace.IsVerboseEnabled)
+                {
+                    EqtTrace.Verbose("DotnetDataCollectionLauncher: File {0}, doesnot exist", runtimeConfigPath);
+                }
+            }
+
+            // Use the deps.json for test source
+            var depsFilePath = Path.Combine(currentWorkingDirectory, string.Concat(dataCollectorFileName, ".deps.json"));
+            if (this.fileHelper.Exists(depsFilePath))
+            {
+                var argsToAdd = " --depsfile \"" + depsFilePath + "\"";
+                args += argsToAdd;
+                if (EqtTrace.IsVerboseEnabled)
+                {
+                    EqtTrace.Verbose("DotnetDataCollectionLauncher: Adding {0} in args", argsToAdd);
+                }
+            }
+            else
+            {
+                if (EqtTrace.IsVerboseEnabled)
+                {
+                    EqtTrace.Verbose("DotnetDataCollectionLauncher: File {0}, doesnot exist", depsFilePath);
+                }
+            }
+
+            commandLineArguments.Insert(0, args);
+
+            var argumentsString = string.Join(" ", commandLineArguments);
+
+            this.DataCollectorProcess = this.processHelper.LaunchProcess(currentProcessFileName, argumentsString, currentWorkingDirectory);
+            return this.DataCollectorProcess.Id;
+        }
+
+        /// <summary>
+        /// Get full path for the .net host
+        /// </summary>
+        /// <returns>Full path to <c>dotnet</c> executable</returns>
+        /// <remarks>Debuggers require the full path of executable to launch it.</remarks>
+        private string GetDotnetHostFullPath()
```

Comment:
This functionality is DRY violation with DotnetTestHostManager. Please extract it to an independent unit.